app/components/app_vaccination_record_details_component.rb

```diff
@@ -9,6 +9,28 @@ def initialize(vaccination_record)
     @batch = vaccination_record.batch
   end
 
+  def dose_number
+    return nil if @vaccine.nil? || @vaccine.seasonal?
+
+    numbers_to_words = {
+      1 => "First",
+      2 => "Second",
+      3 => "Third",
+      4 => "Fourth",
+      5 => "Fifth",
+      6 => "Sixth",
+      7 => "Seventh",
+      8 => "Eighth",
+      9 => "Ninth"
+    }.freeze
+
+    if @vaccination_record.dose_sequence <= 9
+      numbers_to_words[@vaccination_record.dose_sequence]
```

Comment:
Did you remember/consider `ordinalize`?

```ruby
1.ordinalize     # => "1st"
2.ordinalize     # => "2nd"
```

Reply:
I did, but the design in the prototype explicitly used words and I don't think Rails has anything out of the box for that, and bringing in a gem just for the above was overkill 🤷